kitsune/users/utils.py

```diff
@@ -98,9 +99,10 @@ def anonymize_user(user):
     profile.fxa_uid = "{user_id}-{uid}".format(user_id=user.id, uid=str(uid))
     profile.save()
 
-    # Deactivate the user and change key information
+    # Change key information, clear the user's inbox/outbox, and deactivate the user.
     user.username = f"user{uid.int}"
     user.email = f"{uid.int}@example.com"
+    clear_inbox_and_outbox(user)
```

Comment:
I feel that adding a method that will perform two `delete` queries adds unnecessary complexity without any benefits. Let's just trigger the `delete`s here